src/FSAL/FSAL_CORTXFS/mds.c

```diff
@@ -125,18 +129,11 @@ extern struct kvsfs_fsal_module KVSFS;
  * 	kvsfs_pnfs_mds_ctx.mds_ds_list must have atleast one elem, i.e. self
  */
 
-enum kvsfs_pNFS_server_role {
-	KVSFS_PNFS_DISABLED = 0,
-	KVSFS_PNFS_DS,
-	KVSFS_PNFS_MDS,
-	KVSFS_PNFS_BOTH
-};
-
 struct kvsfs_pnfs_ds_addr {
 	uint16_t ds_addr_id;
 	uint16_t ds_addr_ref;
 	in_port_t ds_ip_port;
-	sockaddr_t ds_ip_addr;
+	struct sockaddr_in ds_ip_addr;
```

Comment:
Any specific reason for using sockaddr_in which only supports IPv4 instead of sockaddr_t which is not type specific?

Reply:
The address given out by GETDEVICEINFO is of the type device_addr4, so there is no need to parse IPV6 addresses.So, we can keep it as sockaddr_in to keep things simple.